controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -193,6 +193,14 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		nb.Spec.DisableLoadBalancerService = sc.Spec.MultiCloudGateway.DisableLoadBalancerService
 	}
 
+	if sc.Spec.MultiCloudGateway.ExternalPgConfig.Enable {
```

Comment:
you need `nil` check for `sc.Spec.MultiCloudGateway` here as well

Reply:
done